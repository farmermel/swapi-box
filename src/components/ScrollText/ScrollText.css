.movie-info {
  position: relative;
  top: 110px;
  margin: 0;
  background-color: inherit;
  color: yellow;
  font-size: 3em;
  text-align: center; }

.scroll-container {
  font-family: 'legothickregular';
  color: yellow;
  position: relative;
  width: 100%;
  height: 350px;
  bottom: 0;
  font-size: 350%;
  font-weight: bold;
  text-align: justify;
  overflow: hidden;
  transform-origin: 50% 100%;
  transform: perspective(300px) rotateX(25deg); }
  .scroll-container:after {
    position: absolute;
    content: ' ';
    left: 0;
    right: 0;
    top: 0;
    bottom: 30%;
    background-image: linear-gradient(top, black 0%, transparent 100%);
    pointer-events: none; }

.scroll {
  position: absolute;
  top: 100%;
  animation: scroll 15s linear 4s infinite; }

@keyframes scroll {
  0% {
    top: 100%; }
  100% {
    top: -170%; } }
