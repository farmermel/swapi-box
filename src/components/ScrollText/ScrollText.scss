.movie-info {
  background-color: green;
}

.scroll-container {
  font-family: 'legothickregular';
  background-color: #000;
  color: yellow;
  // position: absolute;
  position: relative;
  width: 18em;
  height: 350px;
  bottom: 0;
  left: 50%;
  margin-left: -9em;
  font-size: 350%;
  font-weight: bold;
  text-align: justify;
  overflow: hidden;
  transform-origin: 50% 100%;
  transform: perspective(300px) rotateX(25deg);
  &:after {
    position: absolute;
    content: ' ';
    left: 0;
    right: 0;
    top: 0;
    bottom: 60%;
    background-image: linear-gradient(top, rgba(0,0,0,1) 0%, transparent 100%);
    pointer-events: none;
  }
}

.scroll {
  position: absolute;
  top: 100%;
  animation: scroll 100s linear 4s infinite;
}

@keyframes scroll {
  0% { top: 100%; }
  100% { top: -170%; }
}