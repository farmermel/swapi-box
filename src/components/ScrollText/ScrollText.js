import React from 'react';
import PropTypes from 'prop-types';
import './ScrollText.css';

const ScrollText = ({ movieData }) => {
  return (
    <div>
      <h3 className='movie-info'>
        {movieData.title}
        {movieData.date}
      </h3>
      <article className='scroll-container'>
        <div className='scroll'>
          {movieData.scroll}
        </div>
      </article>
    </div>
  )
};

ScrollText.propTypes = {
  movieData: PropTypes.objectOf(PropTypes.string)
};

export default ScrollText;