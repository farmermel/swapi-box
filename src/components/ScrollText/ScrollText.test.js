import React from 'react';
import { shallow } from 'enzyme';
import ScrollText from './ScrollText';
import mockData from '../../mock-data';

describe('ScrollText', () => {
  it('matches the snapshot', () => {
    const movieData = mockData.cleaned.film;
    let wrapper = shallow(<ScrollText movieData={movieData} />);

    expect(wrapper).toMatchSnapshot();
  });
});