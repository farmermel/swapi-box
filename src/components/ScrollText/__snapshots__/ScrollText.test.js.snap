// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollText
    movieData={
        Object {
            "date": "2017",
            "scroll": "BLAHBLAH STAR WARS",
            "title": "SITH",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="movie-info"
>
          SITH
          2017
</h3>,
        <article
          className="scroll-container"
>
          <div
                    className="scroll"
          >
                    BLAHBLAH STAR WARS
          </div>
</article>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "SITH",
            "2017",
          ],
          "className": "movie-info",
        },
        "ref": null,
        "rendered": Array [
          "SITH",
          "2017",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="scroll"
>
            BLAHBLAH STAR WARS
</div>,
          "className": "scroll-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BLAHBLAH STAR WARS",
            "className": "scroll",
          },
          "ref": null,
          "rendered": "BLAHBLAH STAR WARS",
          "type": "div",
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="movie-info"
>
            SITH
            2017
</h3>,
          <article
            className="scroll-container"
>
            <div
                        className="scroll"
            >
                        BLAHBLAH STAR WARS
            </div>
</article>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "SITH",
              "2017",
            ],
            "className": "movie-info",
          },
          "ref": null,
          "rendered": Array [
            "SITH",
            "2017",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="scroll"
>
              BLAHBLAH STAR WARS
</div>,
            "className": "scroll-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BLAHBLAH STAR WARS",
              "className": "scroll",
            },
            "ref": null,
            "rendered": "BLAHBLAH STAR WARS",
            "type": "div",
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
